app/controllers/api/storage.php

```diff
@@ -1018,12 +1031,21 @@
             );
         }
 
-        if (!empty($file->getAttribute('algorithm', ''))) {
-            if (empty($source)) {
-                $source = $deviceFiles->read($path);
-            }
-            $compressor = new GZIP();
-            $source = $compressor->decompress($source);
+        switch ($file->getAttribute('algorithm', 'none')) {
+            case 'zstd':
+                if (empty($source)) {
+                    $source = $deviceFiles->read($path);
+                }
+                $compressor = new Zstd();
+                $source = $compressor->decompress($source);
+                break;
+            case 'gzip':
+                if (empty($source)) {
+                    $source = $deviceFiles->read($path);
+                }
+                $compressor = new GZIP();
+                $source = $compressor->decompress($source);
+                break;
```

Comment:
Dont we need a default case?  
Because when we compress, we are defaulting to GZIP 
https://github.com/appwrite/appwrite/pull/3737/files#diff-02d3b2d220ab36c4733052fd1e3e017df738b47f7b3033dc8a93efbb7752436bR513

Reply:
In those cases we are already checking algorithm is not equal to `none` and having default there to gzip to keep it consistent if people upgrade as previously we were not exposing it but were using gzip compression and when we upgrade we can set it to gzip, but yes we can remove the default case in above methods where you pointed if we want to, and make sure when we migrate, we set `compression` to `gzip`. Let me know what you think?

Reply:
As a general idea, its always good to have a default case in every switch case. I prefer we keep a default for the sake of readability so anyone who reads it in the future knows what the logic is